server/routes/createGoal/createGoalController.ts

```diff
@@ -75,17 +76,26 @@ export default class CreateGoalController {
     const targetDate = parseDate(req, 'targetDate')
     req.session.addStepForm = { ...req.body, targetDate }
     const { addStepForm } = req.session
+    const { addStepForms } = req.session
 
     const errors = validateAddStepForm(addStepForm)
     if (errors.length > 0) {
       req.flash('errors', errors)
       return res.redirect(`/plan/${prisonNumber}/goals/add-step`)
     }
 
+    // check to see if this step has already been added (e.g. after the user clicks the back button)
+    const existingAddStepForm = addStepForms.find(step => step.stepNumber === addStepForm.stepNumber)
+    if (!existingAddStepForm) {
+      addStepForms.push(req.session.addStepForm)
+    } else {
+      // update it in case the user has clicked back and changed it
+      existingAddStepForm.title = addStepForm.title
```

Comment:
Based on your unit tests I can see this works, though it surprises me slightly because theres no obvious "update the original object / array element"

I guess this tells us something about how TS/JS works .... in that it looks like variables are passed by reference rather than by value ..... perhaps something to watch out for in the future .... 👀 

(Everyday's a school day! 🤣 )